src/helpers/length.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Length function for FGH
+ *
+ * Implements the JQ length function which returns:
+ * - For strings: the number of Unicode codepoints
+ * - For numbers: the absolute value
+ * - For arrays: the number of elements
+ * - For objects: the number of key-value pairs
+ * - For null: zero
+ * - For booleans: error
+ */
+
+/**
+ * Get the length of a value according to JQ rules
+ *
+ * @param x The value to determine the length of
+ * @returns The length as a number
+ * @throws Error when calculating the length of a boolean
+ */
+export const getLength = (x: unknown): number[] => {
+  if (Array.isArray(x)) {
+    const len = x.length
+    const result = new Array(len)
```

Comment:
I'd use `result = x.map(i => getLengthOfSingleValue(i))`

Reply:
And it would be at least an order of magnitude slower.